src/openforms/registrations/contrib/objects_api/config.py

```diff
@@ -1,13 +1,57 @@
+from typing import Any
+
 from django.utils.translation import gettext_lazy as _
 
+from drf_spectacular.utils import OpenApiExample, extend_schema_serializer
 from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
 
+from openforms.formio.validators import variable_key_validator
 from openforms.template.validators import DjangoTemplateValidator
 from openforms.utils.mixins import JsonSchemaSerializerMixin
 from openforms.utils.validators import validate_rsin
 
 
+@extend_schema_serializer(
+    examples=[
+        OpenApiExample(
+            name="Variable mapping example",
+            value={
+                "variable_key": "my_edit_grid.0.field",
+                "target_path": ["path", "to.the", "target"],
+            },
+        )
+    ]
+)
+class ObjecttypeVariableMappingSerializer(serializers.Serializer):
+    """A mapping between a form variable key and the corresponding Objecttype JSON location."""
+
+    variable_key = serializers.CharField(
+        label=_("variable key"),
+        help_text=_(
+            "The 'dotted' path to a form variable key. The format should comply to how Formio handles nested component keys."
+        ),
+        validators=[variable_key_validator],
+    )
+    target_path = serializers.ListField(
+        child=serializers.CharField(label=_("Part of a JSON path")),
+        label=_("target path"),
+        help_text=_(
+            "Representation of the JSON target location as a list of string 'bits'."
+        ),
+    )
+
+
 class ObjectsAPIOptionsSerializer(JsonSchemaSerializerMixin, serializers.Serializer):
+
+    version = serializers.ChoiceField(
+        label=_("options version"),
+        help_text=_(
+            "The schema version of the objects API Options. Not to be confused with the objecttype version."
+        ),
+        choices=[1, 2],
+        default=1,
```

Comment:
I'd use a `models.IntegerChoices` enum for this so that you can add proper typing support and have exhaustiveness checks when processing the possible options versions during registration.